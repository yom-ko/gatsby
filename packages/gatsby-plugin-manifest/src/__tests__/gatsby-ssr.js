// const { shallow } = require(`enzyme`)
const { onRenderBody } = require(`../gatsby-ssr`)

let headComponents
const setHeadComponents = args => (headComponents = headComponents.concat(args))

const ssrArgs = {
  setHeadComponents,
}

describe(`gatsby-plugin-manifest`, () => {
  beforeEach(() => {
    global.__PATH_PREFIX__ = ``
    headComponents = []
  })

  it(`Adds "shortcut icon" and "manifest" links and "theme_color" meta tag to head`, () => {
    onRenderBody(ssrArgs, { icon: true, theme_color: `#000000` })
    expect(headComponents).toMatchSnapshot()
  })

  it(`Does not add a "theme_color" meta tag to head if "theme_color" option is not provided or is an empty string`, () => {
    onRenderBody(ssrArgs, { icon: true })
    expect(headComponents).toMatchSnapshot()
  })

  describe(`Creates legacy apple touch links if opted in`, () => {
    it(`Using default set of icons`, () => {
      onRenderBody(ssrArgs, {
        icon: true,
        theme_color: `#000000`,
        legacy: true,
      })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Using user specified list of icons`, () => {
      onRenderBody(ssrArgs, {
        icon: true,
        theme_color: `#000000`,
        legacy: true,
        icons: [
          {
            src: `/favicons/android-chrome-48x48.png`,
            sizes: `48x48`,
            type: `image/png`,
          },
          {
            src: `/favicons/android-chrome-512x512.png`,
            sizes: `512x512`,
            type: `image/png`,
          },
        ],
      })
      expect(headComponents).toMatchSnapshot()
    })
  })

  it(`Creates href attributes using pathPrefix`, () => {
    global.__PATH_PREFIX__ = `/path-prefix`

    onRenderBody(ssrArgs, {
      icon: true,
      theme_color: `#000000`,
      legacy: true,
    })

    headComponents
      .filter(component => component.type === `link`)
      .forEach(component => {
        expect(component.props.href).toEqual(
          expect.stringMatching(/^\/path-prefix\//)
        )
      })
  })
})
