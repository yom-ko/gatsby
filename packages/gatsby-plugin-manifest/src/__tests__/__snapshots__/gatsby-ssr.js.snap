// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-manifest Adds "shortcut icon" and "manifest" links and "theme_color" meta tag to head 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
]
`;

exports[`gatsby-plugin-manifest Creates legacy apple touch links if opted in Using default set of icons 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
  <link
    href="/icons/icon-48x48.png"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Creates legacy apple touch links if opted in Using user specified list of icons 1`] = `
Array [
  <link
    href="/favicons/android-chrome-48x48.png"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
  <link
    href="/favicons/android-chrome-48x48.png"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/favicons/android-chrome-512x512.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Does not add a "theme_color" meta tag to head if "theme_color" option is not provided or is an empty string 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
]
`;
